analyzer/consensus/consensus.go

```diff
@@ -191,6 +202,57 @@ func (m *Main) latestBlock(ctx context.Context) (int64, error) {
 	return latest, nil
 }
 
+func (m *Main) isGenesisProcessed(ctx context.Context) (bool, error) {
+	var processed bool
+	if err := m.target.QueryRow(
+		ctx,
+		m.qf.IsGenesisProcessedQuery(),
+		m.cfg.ChainID,
+		consensusDamaskAnalyzerName,
+	).Scan(&processed); err != nil {
+		return false, err
+	}
+	return processed, nil
+}
+
+func (m *Main) processGenesis(ctx context.Context) error {
+	m.logger.Info("fetching genesis document")
+	genesisDoc, err := m.cfg.Source.GenesisDocument(ctx)
+	if err != nil {
+		return err
+	}
+
+	m.logger.Info("processing genesis document")
+	gen := NewMigrationGenerator(m.logger.With("height", "genesis"))
+	buf := new(bytes.Buffer)
+	if err := gen.WriteGenesisDocumentMigrationOasis3(buf, genesisDoc); err != nil {
+		return err
+	}
+	sql := buf.String()
+
+	// Debug: log the SQL into a file if requested.
+	debugPath := os.Getenv("CONSENSUS_DAMASK_GENESIS_DUMP")
+	if debugPath != "" {
+		if err := os.WriteFile(debugPath, []byte(sql), 0o644); err != nil {
```

Comment:
(non-blocking) `0o644` potentially helpful with a comment.

Reply:
I think 644 and 755 are very common and recognizable modes. But it can't hurt to explain; comment added.

Reply:
I'm searching around, it it looks like it's normal to use 666 and let umask take care of lowering the group+other permissions as needed

https://pkg.go.dev/os#Create Create uses 666 by default, for example

Reply:
This is just a debug file so I really don't worry about it much. Changed to 666 though.

Reply:
Ahaha now the linter wants `0o600`. I would slightly prefer 644 or more permissive in case this is being run from docker and the file ends up being owned by root ... but that's pretty hypothetical, I don't feel strongly enough to create a lint exception. All bow to the linter. 600 it is.

Reply:
what did it say about 0o600?